### PR TITLE
feat: add `pkg-types` to support indentation in templates

### PR DESCRIPTION
**Pull request by sadeghbarati**

### CODE REVIEW

File: packages/cli/index.ts [hunk 1]
```diff
@@ -115,14 +115,14 @@ const main = defineCommand({
 
           for (const p of paths) {
             const pJsonPath = path.resolve(p, "package.json");
-            const { name } = await importPackageJson(pJsonPath);
+            const { name } = await readPackageJSON(pJsonPath);
 
-            if (compact) {
+            if (compact && name) {
```
Comment (Aslemammad) [inline packages/cli/index.ts:120]: let's check the name in another condition

**Review by Aslemammad — CHANGES_REQUESTED** [review]

File: packages/cli/index.ts (hunk 1 above)
Comment (sadeghbarati) [reply packages/cli/index.ts]: Done 🙌

File: packages/cli/index.ts [hunk 2]
```diff
@@ -115,14 +115,16 @@ const main = defineCommand({
...
             if (compact) {
-              await verifyCompactMode(name);
+              if(name) {
+                await verifyCompactMode(name);
+              }
             }
 
             deps.set(
-              name,
+              name!,
```
Comment (sadeghbarati) [inline packages/cli/index.ts:127]: Should I have to do the same for this too? (check it name is defined)

Comment (Aslemammad) [reply packages/cli/index.ts:127]: no just on line 119 put a condition and throw an error if it's not defined
so we use one condition instead of two

Comment (sadeghbarati) [reply packages/cli/index.ts]: Done

File: packages/cli/index.ts [hunk 3]
```diff
@@ -115,7 +115,10 @@ const main = defineCommand({
 
           for (const p of paths) {
             const pJsonPath = path.resolve(p, "package.json");
-            const { name } = await importPackageJson(pJsonPath);
+            const { name } = await readPackageJSON(pJsonPath);
+            if(!name) {
+              throw new Error('name is not defined');
```
Comment (Aslemammad) [inline packages/cli/index.ts:120]: better to write: `"name" field in ${pJsonPath` should be defined

**Comment by sadeghbarati** [comment]
This may take time to resolve conflicts correctly

**Pull request closed without merging**